volttron/platform/web/admin_endpoints.py

```diff
@@ -267,24 +263,37 @@ def __delete_csr_api(self, common_name):
         return Response(jsonapi.dumps(data), content_type="application/json")
 
     def __pending_csrs_api(self):
-        csrs = [c for c in self._certs.get_pending_csr_requests()]
-        return Response(jsonapi.dumps(csrs), content_type="application/json")
+        try:
+            data = self._rpc_caller.call(AUTH, 'get_pending_csrs').get(timeout=4)
+
+        except TimeoutError as e:
+            data = dict(status="ERROR", message=e.message)
+
+        return Response(jsonapi.dumps(data), content_type="application/json")
 
     def __cert_list_api(self):
 
-        subjects = [dict(common_name=x.common_name)
-                    for x in self._certs.get_all_cert_subjects()]
-        return Response(jsonapi.dumps(subjects), content_type="application/json")
+        try:
+            data = [dict(common_name=x.common_name) for x in
+                    self._rpc_caller.call(AUTH, "get_all_pending_csr_subjects").get(timeout=2)]
+
+        except TimeoutError as e:
+            data = dict(status="ERROR", message=e.message)
+
+        return Response(jsonapi.dumps(data), content_type="application/json")
 
     def __approve_credential_api(self, user_id):
         try:
             _log.debug("Creating credential and permissions for user: {}".format(user_id))
-            self._rpc_caller.call(AUTH, 'approve_authorization_failure', user_id).wait()
+            self._rpc_caller.call(AUTH, 'approve_authorization_failure', user_id).wait(timeout=4)
```

Comment:
why wait() instead of get()

Reply:
Wait is blocking, but does not return a value. Get is used for return values. "approve_authorization_failure" does not have a return value.